internal/scheduler/queue/queue_cache.go

```diff
@@ -72,20 +73,44 @@ func (c *ApiQueueCache) fetchQueues(ctx *armadacontext.Context) error {
 		return err
 	}
 	queues := make([]*api.Queue, 0)
-	for {
-		msg, err := stream.Recv()
-		if err != nil {
-			return err
+	shouldEndStream := false
+
+	for !shouldEndStream {
+		msg, recvErr := stream.Recv()
+		if recvErr != nil {
+			return recvErr
 		}
 		switch msg.GetEvent().(type) {
 		case *api.StreamingQueueMessage_Queue:
 			queues = append(queues, msg.GetQueue())
 		case *api.StreamingQueueMessage_End:
 			c.queues.Store(&queues)
 			ctx.Infof("Refreshed Queues in %s", time.Since(start))
-			return nil
+			shouldEndStream = true
 		default:
 			return fmt.Errorf("unknown event of type %T", msg.GetEvent())
 		}
 	}
+
+	if closeErr := closeStream(stream); closeErr != nil {
+		return fmt.Errorf("closing get queues stream: %w", closeErr)
+	}
+
+	return nil
+}
+
+// This should be called after our end of stream message has been seen (StreamingQueueMessage_End)
+// We call recv one more time and expect an EOF back, indicating the stream is properly closed
+func closeStream(stream api.Submit_GetQueuesClient) error {
+	res, err := stream.Recv()
+	if err == nil {
+		switch typed := res.GetEvent().(type) {
+		default:
+			return fmt.Errorf("failed closing stream - unexpectedly received event of type %T", typed)
```

Comment:
Isn't it the same if you change `typed` to `res.GetEvent()` and remove the whole `switch` statement?

Reply:
This code was actually copied from another part of Armada that has solved this issue already: https://github.com/armadaproject/armada/blob/08f6f8a690d10a7370ee63944afb65529656e043/internal/executor/service/lease_requester.go#L104-L118

Reply:
I always prefer to keep convention but I'm happy to change both places.